packages/dev/core/src/Meshes/mesh.vertexData.subdivide.ts

```diff
@@ -0,0 +1,457 @@
+import { Vector3 } from "core/Maths/math.vector";
+import { VertexData } from "./mesh.vertexData";
+import { Scalar } from "core/Maths/math.scalar";
+import type { FloatArray } from "core/types";
+
+/**
+ * Inspired by https://github.com/stevinz/three-subdivide
+ * Thanks a lot to https://github.com/stevinz
+ */
+
+/**
+ * Interface used to configure the subdivision process
+ */
+export interface ISubdivideOptions {
+    /** Apply only flat subdivision - false by default */
+    flatOnly?: boolean;
+    /** Split all triangles at edges shared by coplanar triangles - true by default*/
+    split?: boolean;
+    /**  Should UV values be averaged during subdivision - false by default */
+    uvSmooth?: boolean;
+    /** Should edges / breaks in geometry be ignored during subdivision? - false by default */
+    preserveEdges?: boolean;
+    /** How much to weigh favoring heavy corners vs favoring Loop's formula - 1 by default*/
+    weight?: number;
+}
+
+const _positionShift = Math.pow(10, 4);
+
+/**
+ * Rounds a number (simulate integer rounding)
+ * @internal

```

Comment:
nit: I don't think you need the "@internal" decorator in all these functions, as they are not exported?

Reply:
The system complains because not all the parameters are not documented (hence internal)